cds/modules/migrator/tasks.py

```diff
@@ -71,3 +83,137 @@ def clean_record(data, source_type):
     except Exception:
         db.session.rollback()
         raise
+
+
+@shared_task(ignore_result=True)
+def check_record(data, source_type):
+    """Verify if the record and files were correctly migrated."""
+    source_type = source_type or 'marcxml'
+    assert source_type in ['marcxml', 'json']
+
+    recorddump = current_migrator.records_dump_cls(
+        data,
+        source_type=source_type,
+        pid_fetchers=current_migrator.records_pid_fetchers, )
+    recorddump.prepare_revisions()
+    recorddump.prepare_pids()
+
+    # First verify if the record exists
+    recid_pid = PersistentIdentifier.query.filter(
+        PersistentIdentifier.pid_value == recorddump.recid,
+        PersistentIdentifier.pid_type == 'recid').one_or_none()
+    if not recid_pid:
+        logger.error('PID not found: {0}'.format(recorddump.recid))
+        raise Exception('Record {0} not migrated'.format(recorddump.recid))
+
+    record = Record.get_record(recid_pid.object_uuid)
+    # The record exists
+    assert record
+
+    _chec_files(record, recorddump)
+    _check_web(record)
+    _check_es(record)
+
+
+def _md5(fname):
+    hash_md5 = hashlib.md5()
+    with open(fname, "rb") as f:
+        for chunk in iter(lambda: f.read(4096), b""):
+            hash_md5.update(chunk)
+    return hash_md5.hexdigest()
+
+
+def _check_web(record):
+    """Check record accessibility."""
+    urls = [
+        "https://videos.cern.ch/record/{recid}",
+        "https://videos.cern.ch/api/record/{recid}"
+    ]
+    if 'video-v1.0.0.json' in record['$schema']:
+        urls.extend([
+            "https://videos.cern.ch/video/{rn}",
```

Comment:
what about if it is a project? Will it work?

Reply:
this two urls are for the embed and media export, the url will work but it's useless so I guess we can leave like that